lms/services/lti_outcomes.py

```diff
@@ -1,26 +1,74 @@
+from datetime import datetime, timezone
 from xml.parsers.expat import ExpatError
 
 import xmltodict
 
+from lms.services import LTIAHTTPService
 from lms.services.exceptions import ExternalRequestError
 
 __all__ = ["LTIOutcomesClient"]
 
 
 class LTIOutcomesClient:
```

Comment:
I have a suggestion here which is pretty disruptive. I don't think this is one class, I think it's two which have been put together.

I think the correct solution here might be do use a feature of the services we don't make use of normally, which is the ability to send back a difference class from the interface. Something like:

```python
class LTIGradingClient:
    def read_result(self, grading_id):
        ...

    def record_result(self, grading_id, score=None, pre_record_hook=None):
        ...


class LTI13GradingClient(LTIGradingClient):
    ...


class LTI11GradingClient(LTIGradingClient):
    ...
```

Then in the factory:

```python
def factory(_context, request):
    application_instance = request.find_service(name="application_instance").get_current()
    if application_instance.lti_version == "1.3.0":
        return LTI13GradingClient(...)
    else:
        return LTI11GradingClient(...)
```

I think this would have a few positive outcomes:

 * Each class becomes about one thing only and can be split into sub-files in the dir
 * It's clear what items are for supporting 1.1, 1.3 or are shared (probably very little)
 * All the switching behavior is removed
 * The LTI 1.3 class will probably get to a size where `LTIAHTTPService` can either become private to it, or inlined